src/coreclr/jit/ssabuilder.cpp

```diff
@@ -550,6 +603,31 @@ void SsaBuilder::InsertPhi(BasicBlock* block, unsigned lclNum)
     JITDUMP("Added PHI definition for V%02u at start of " FMT_BB ".\n", lclNum, block->bbNum);
 }
 
+#if defined(TARGET_WASM)
+void SsaBuilder::InsertPhiToRationalIRForm(BasicBlock* block, unsigned lclNum)
+{
+    assert(m_pCompiler->compRationalIRForm);
+
+    var_types type = m_pCompiler->lvaGetDesc(lclNum)->TypeGet();
+
+    // PHIs and all the associated nodes do not generate any code so the costs are always 0
+    GenTree* phi = new (m_pCompiler, GT_PHI) GenTreePhi(type);
+    phi->SetCosts(0, 0);
+    GenTree* storeLcl = (GenTree*)m_pCompiler->gtNewStoreLclVar(lclNum, phi);
```

Comment:
```suggestion
    GenTree* phi      = new (m_pCompiler, GT_PHI) GenTreePhi(type);
    GenTree* storeLcl = m_pCompiler->gtNewStoreLclVar(lclNum, phi);
```
LIR doesn't have a way to maintain costs anyway, so, no point setting them.